src/ua_types_encoding_binary.c

```diff
@@ -1051,6 +1081,50 @@ Variant_decodeBinaryUnwrapExtensionObject(UA_Variant *dst, Ctx *ctx) {
     return decodeBinaryJumpTable[dst->type->typeKind](dst->data, dst->type, ctx);
 }
 
+static status
+Variant_decodeBinaryUnwrapArrayOfExtensionObject(UA_Variant *dst, Ctx *ctx) {
+    size_t length = 0;
+    status ret = Array_decodeBinaryLength(&dst->data, &length, ctx);
+    UA_CHECK(ret == UA_STATUSCODE_GOOD && length > 0, return ret);
+
+    uintptr_t ptr;
+    UA_Boolean first = true;
+    const UA_DataType *type;
+    for(size_t i = 0; i < length; ++i) {
+
+        /* Decode the type */
+        ret = ExtensionObject_decodeBinaryTypeId(&type, ctx);
+        UA_CHECK_STATUS(ret,
+                        if(dst->data) {
+                            UA_Array_delete(dst->data, i, dst->type);
+                            dst->data = NULL;
+                        }
+                        return ret);
+
+        if(first) {
```

Comment:
Thanks for this note, we didn't think about that. However, even if it is a possible case by the specification it feels like a strange use case to have an array with elements of different types. Do you have a valid use-case where this mixed array is used or is it only a theoretical case? 

If it is just a theoretical case I am thinking that it could be ok to not support it and fail the decoding if any processed element has a type that differs from the type of the first element in the array.

Reply:
Even if there is no immediate use case, it is possible to write "mixed type" arrays to a variable.
For example by setting the variable's type to "ExtensionObject" and then filling it with whatever.

Even if this is a synthetic case, we need to remain 100% feature complete when it comes to the expressive power of the protocol. So a first pass should detect "mixed type" arrays and then fallback to the ExtensionObject wrapper.

Reply:
This comment should now be solved, also added a unittest to verify it.